nuclear/dependencies.py

```diff
@@ -0,0 +1,85 @@
+import os
+import re
+import subprocess
+import sys
+
+whitespace = re.compile("\s+")
+
+
+def parse_requirements_map(user_tools_path):
+    file_path = os.path.abspath(os.path.join(user_tools_path, "requirements.map"))
+    requirements_map = {}
+
+    with open(file_path, "r") as file:
+        for line in file:
+            line = line.strip()
+
+            if len(line) == 0 or line.startswith("#"):
+                continue
+
+            import_name, package_name = whitespace.split(line.strip())
+            requirements_map[import_name] = package_name
+
+    return requirements_map
+
+
+# https://regexr.com/5le57
+dependency_name_regex = re.compile(r"^(.+?)(===|==|~=|!=|<=|>=|>|<)")
+
+
+def parse_requirement_versions(user_tools_path):
+    file_path = os.path.abspath(os.path.join(user_tools_path, "requirements.txt"))
+    versions = {}
+
+    with open(file_path, "r") as file:
+        for line in file:
+            line = line.strip()
+
+            if len(line) == 0 or line.startswith("#"):
+                continue
+
+            match = dependency_name_regex.match(line)
+
+            if match:
+                versions[match.group(1)] = line
+
+    return versions
+
+
+dependency_map = None
+
+
+def create_dependency_map(user_tools_path):
+    requirements_map = parse_requirements_map(user_tools_path)
+    versions = parse_requirement_versions(user_tools_path)
+
+    global dependency_map
```

Comment:
This pulls the variable `dependency_map` from above into the local scope so we can write to it. `dependency_map` is defined outside in the first place because we use it in both `parse_requirements_map` and `parse_requirement_versions`, and we don't want to parse the dependencies more than once (since this requires reading two files).

Reply:
`dependency_map` isn't used in either of those functions?

Is reading two small files that big an overhead? I get that this might avoid doing it more than once, but that seems kind of trivial...
It's not a big deal obviously, but this just seems a little convoluted to me

Reply:
man, I am nitpicking too hard... Imma just resolve my remaining ones lol.